### DISCUSSION TITLE
PIO: fatal error: stm32_def.h: No such file or directory

### DISCUSSION THREAD

**Discussion by khoih-prog** [khoih-prog/AsyncMQTT_Generic#14] [Q&A]
Moved from  #3, original from @fsnyders

---

Hello @khoih-prog ,

I am struggling to build the FullyFeaturedSSL_ESP32 file on my ESP32 dev. module. I am using PlatformIO and I have installed the AsyncMQTT_Generic@^1.7.0 library. Somehow I can't resolve the following error: 

```
.pio/libdeps/esp32dev/STM32Ethernet/src/utility/stm32_eth.h:42:10: fatal error: stm32_def.h: No such file or directory .
```

Any advice would be greatly appreciated, thanks!

**Comment by khoih-prog** [comment]
Use the latest version [**AsyncMQTT_Generic** v1.7.0](https://github.com/khoih-prog/AsyncMQTT_Generic/releases/tag/v1.7.0). If still having problem, add to `platformio.ini`

```
lib_ignore =
   STM32AsyncTCP
   STM32duino LwIP
   STM32Ethernet
```

**Reply by fsnyders** [reply]
Thank you. I have applied these recommendations but I am now experiencing multiple errors with the Portenta_H7_AsyncTCP.cpp and .h files?

**Comment by khoih-prog** [comment]
I suggest you use the Arduino IDE first as `PIO` requires more knowledge and skills to manually modify the settings, especially for multi-platform libraries, such as this [**AsyncMQTT_Generic**](https://github.com/khoih-prog/AsyncMQTT_Generic).

Just continue ignore manually as many libraries as PIO complains. Strange that `PIO` still can't learn from `old stable Arduino IDE` to manage to know, use and request / compile only libraries for the **selected architecture** (such as ESP32 in your case) ???

Go to PIO Forum and search for `stm32_def.h` to see there are some threads to check, such as [**Missing File stm32_eth.cpp.o in Build**](https://community.platformio.org/t/missing-file-stm32-eth-cpp-o-in-build/24425)

Post your issue on [**PIO Forum**](https://community.platformio.org/) for help as this is not the problem of this library.

**Reply by fsnyders** [reply]
Noted. Thank you for the feedback.

**Comment by otl1** [comment]
Hello @khoih-prog  and @fsnyders, I'm also facing the same issue and I don't want to shift to Arduino IDE. I'm new to platformio, If you could suggest any fix I'd really appreciate it. I'm unable to use Async webserver and async mqtt at a time, and getting error - multiple definitions...
